pyproject.toml

```diff
@@ -0,0 +1,123 @@
+[build-system]
+requires = ["setuptools>=62", "wheel"]
```

Comment:
-1: I think this change also affect to users. This requires users who try to install Sphinx via source package must upgrade setuptools to the newer one. For example, the HEAD of Ubuntu still bundles 44.1.
https://packages.ubuntu.com/impish/python-setuptools

Reply:
> This requires users who try to install Sphinx via source package must upgrade setuptools to the newer one.

Local builds (e.g. `pip install .`) now default to isolated environments, which download the build requirements from PyPI, so this wouldn't be an issue, per se.

The bigger issue is that setuptools 62 does not support Python 3.6 at all -- I could re-target this PR to master, but we would then have two different build systems for 12 months (whilst the 5.x branch is active, still supporting Python 3.6) -- not a good solution.

A